arduino-ide.sh

```diff
@@ -2,9 +2,12 @@
 
 FLAGS=''
 
-if [[ $XDG_SESSION_TYPE == "wayland" ]] && [ -c /dev/nvidia0 ]
-then
-    FLAGS="$FLAGS --disable-gpu-sandbox"
+if [[ $XDG_SESSION_TYPE == "wayland" ]]; then
+    if [ -c /dev/nvidia0 ]; then
+        FLAGS="$FLAGS --disable-gpu-sandbox"
+    elif [ -S "$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY" ]; then
+        FLAGS="$FLAGS --enable-features=UseOzonePlatform --ozone-platform=wayland"
```

Comment:
My Only concern with enabling wayland is what are the odds that it is going to break in the future? as previously mentioned wayland + electron are not best friends and i want to ensure that in future updates (if this is merged) wayland support doesnt get completely borked.

Reply:
I'm not sure if this is my system (ill double check on another) but i have also noticed that my cursor theme changes when i enter the arduino ide window when using wayland.

Reply:
Also if this is possible for us to fix here (basically as long as it isnt an upstream issue which i dont think it is) i am getting this error which i think might be an easy fix but i am unsure:
`[55:0930/153033.099125:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is egl, ANGLE is `.

Reply:
> I'm not sure if this is my system (ill double check on another) but i have also noticed that my cursor theme changes when i enter the arduino ide window when using wayland.

That's a bug in org.freedesktop.Sdk 22.08: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1484
It's marked as fixed, so I guess this will go away with the next update.

Reply:
> > I'm not sure if this is my system (ill double check on another) but i have also noticed that my cursor theme changes when i enter the arduino ide window when using wayland.
> 
> That's a bug in org.freedesktop.Sdk 22.08: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1484 It's marked as fixed, so I guess this will go away with the next update.

I figured that might have been the case which is why i wasnt too worried.
The error i posted do you know if that is something we can do about it or are we just going to have to ignore it?
It doesn't prevent anything from working but it would be nice if we can resolve it.

Reply:
> My Only concern with enabling wayland is what are the odds that it is going to break in the future? as previously mentioned wayland + electron are not best friends and i want to ensure that in future updates (if this is merged) wayland support doesnt get completely borked.

@sebastian-de @hfiguiere If you both dont think that there will be a high risk of wayland breaking in the future then i will merge this pr.

Reply:
> I get this error line, too. I had this issue with another electron application, but I can't recall which one. This went away after an electron update.
> 
> However, I now get a segmentation fault, when I start after removing ~/.var/app/cc.arduino.IDE2
> 
> So I guess this really introduces more Problems than it solves. I'll do some more tests, but maybe it's saver to postpone this change.

What is the error?

Reply:
> Also if this is possible for us to fix here (basically as long as it isnt an upstream issue which i dont think it is) i am getting this error which i think might be an easy fix but i am unsure: `[55:0930/153033.099125:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is egl, ANGLE is `.

also regarding this error it doesnt appear to affect the application as it still runs but i'd like to avoid errors if possible. Like you said this could just be because of the older version of electron used.

Reply:
> ```
> flatpak run cc.arduino.IDE2 --log-level=warn
> [31 zypak-helper] Using spawn strategy test 1 as set by environment
> Starting backend process. PID: 31
> [55:0930/172920.853101:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is egl, ANGLE is 
> [55:0930/172920.856926:ERROR:sandbox_linux.cc(374)] InitializeSandbox() called with multiple threads in process gpu-process.
> Using browser-only version of superagent in non-browser environment
> Configuring to accept webviews on '^.+\.webview\..+$' hostname.
> root WARN The local plugin referenced by local-dir:/home/user/.arduinoIDE/plugins does not exist.
> root WARN The local plugin referenced by local-dir:/home/user/.arduinoProIDE/plugins does not exist.
> root WARN The local plugin referenced by local-dir:/home/user/.arduinoIDE/extensions does not exist.
> /app/bin/arduino-ide: line 16:     3 Segmentation fault      (core dumped) env TMPDIR=$XDG_CACHE_HOME zypak-wrapper /app/arduino-ide/arduino-ide /app/arduino-ide/resources/app/scripts/arduino-ide-electron-main.js $FLAGS "$@"
> ```

Do me a favour and try removing the app and then installing it again. It looks like part of the application might be missing.

Reply:
```text
[55:0930/153033.099125:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is egl, ANGLE is
```

I periodically see this message when using Arduino's official builds of the IDE. So its occurrence while you were testing this PR might have been incidental rather than caused by the changes proposed here.

Reply:
> ```
> [55:0930/153033.099125:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is egl, ANGLE is
> ```
> 
> I periodically see this message when using Arduino's official builds of the IDE. So its occurrence while you were testing this PR might have been incidental rather than caused by the changes proposed here.

Ok thanks, this means it might just be the build of the ide that is throwing/causing this error then and because the ide still runs i will ignore it for now. If the error appears in the next release in both the official and flatpak build i might report it then just to see if it is something that they need to look at.